src/components/course/course-header/CourseImportantDates.jsx

```diff
@@ -0,0 +1,137 @@
+  label,
+  children,
+}) => (
+  <Row className="course-important-date border-bottom mx-0 py-2.5">
+    <Col className="px-0">
+      <Stack className="small" direction="horizontal" gap={2}>
+        <Icon size="sm" src={Calendar} />
+        {label}
+      </Stack>
+    </Col>
+    <Col>
+      {children}
+    </Col>
+  </Row>
+);
+
+CourseImportantDate.propTypes = {
+  label: PropTypes.string.isRequired,
+  children: PropTypes.node.isRequired,
+};
+
+const CourseImportantDates = () => {
+  const { courseKey } = useParams();
+  const { data: redeemableLearnerCreditPolicies } = useRedeemablePolicies();
+  const { data: courseMetadata } = useCourseMetadata();
+  const intl = useIntl();
+  const {
+    allocatedCourseRunAssignments,
+    allocatedCourseRunAssignmentKeys,
+    hasAssignedCourseRuns,
+  } = determineAllocatedCourseRunAssignmentsForCourse({
+    redeemableLearnerCreditPolicies,
+    courseKey,
+  });
+
+  const [searchParams] = useSearchParams();
+  const courseRunKey = searchParams.get('course_run_key')?.replaceAll(' ', '+');
+  // Check if the corresponding course run key from query parameters matches an allocated assignment course run key
+  const doesNotHaveCourseRunAssignmentForCourseRunKey = !!courseRunKey && !allocatedCourseRunAssignmentKeys.includes(
+    courseRunKey,
+  );
+  if (!hasAssignedCourseRuns || doesNotHaveCourseRunAssignmentForCourseRunKey) {
+    return null;
+  }
+
+  // Retrieve soonest expiring enroll-by date
+  const { soonestExpirationDate, soonestExpiringAssignment } = getSoonestEarliestPossibleExpirationData({
+    assignments: allocatedCourseRunAssignments,
+    dateFormat: DATETIME_FORMAT,
+  });
+
+  // Match soonest expiring assignment to the corresponding course start date from course metadata
+  let soonestExpiringAllocatedAssignmentCourseStartDate = null;
+  if (soonestExpiringAssignment) {
+    soonestExpiringAllocatedAssignmentCourseStartDate = courseMetadata.availableCourseRuns.find(
+      (courseRun) => courseRun.key === soonestExpiringAssignment?.contentKey,
+    )?.start;
+  }
+
+  // Parse logic of date existence and labels
+  const enrollByDate = soonestExpirationDate ?? null;
+  const courseStartDate = soonestExpiringAllocatedAssignmentCourseStartDate
+    ? dayjs(soonestExpiringAllocatedAssignmentCourseStartDate).format(DATE_FORMAT)
+    : null;
+  const courseHasStartedLabel = hasCourseStarted(courseStartDate)
+    ? intl.formatMessage(messages.courseStarted)
+    : intl.formatMessage(messages.courseStarts);
+
+  if (!enrollByDate && !courseStartDate) {
+    return null;
+  }
+
+  return (
+    <section className="assignments-important-dates mt-4">
+      <h3 className="mb-3">
+        {intl.formatMessage(messages.importantDates)}
+      </h3>
+      {enrollByDate && (
+        <CourseImportantDate label={intl.formatMessage(messages.enrollByDate)}>
+          <small>
+            {enrollByDate}
+          </small>
+        </CourseImportantDate>
+      )}
```

Comment:
**Current**

![image](https://github.com/user-attachments/assets/f6768e30-ff3e-425b-ab03-0ca769e6ac41)

1. Note how the dates are not exactly vertically aligned with the corresponding "Enroll-by date" / "Course starts" label.
2. There's a bit excessive spacing below the `h3` and the `CourseImportantDate`; I would rely on the default `margin-bottom` that comes with the `h3` (i.e., remove `.mb-3`).

**Proposed/Expected**

1. Removed `.mb-3` on `h3`.
2. Consolidating the `small` typography styles by removing the `small` `className` within the `CourseImportantDate` sub-component's `Stack` and the `<small>` elements, in favor of just passing `small` on the `<section className="assignments-important-dates mt-4 small">`.

![image](https://github.com/user-attachments/assets/fc392c03-afb0-4778-be81-186b694b671f)